tests/unit/semaphore_test.cc

```diff
@@ -426,6 +427,19 @@ SEASTAR_THREAD_TEST_CASE(test_semaphore_abort_after_wait) {
     BOOST_REQUIRE_EQUAL(x, 0);
 }
 
+SEASTAR_THREAD_TEST_CASE(test_semaphore_abort_with_exception_after_wait) {
+    auto sem = semaphore(0);
+    abort_source as;
+    int x = 0;
+    auto fut1 = sem.wait(as).then([&x] {
+        x++;
+    });
+    as.request_abort_ex(std::runtime_error("expected"));
+    sem.signal();
+    BOOST_CHECK_THROW(fut1.get(), std::runtime_error);
```

Comment:
How confident are we, that this thrown-and-caught `runtime_exception` is the one from above `as.request_abort_ex()`, but not some "generic" runtime error that accidentally crept in along the way?

Reply:
I have no reason to suspect that, but I can change this to a unique exception type to increase the confidence.

Reply:
Yes, please

Reply:
done in v2